workflow-templates/im-deploy-az-database.yml

```
@@ -0,0 +1,176 @@
+        required: true
+      environment:
+        # TODO: Update this description with the correct environments
+        description: 'The environment to deploy to - dev|qa|stage|prod|demo|uat'
+        required: true
+env:
+  GITHUB_REF: ${{ github.event.inputs.branch-tag-sha  }} # This is the branch/tag/sha that we'll be deploying
+  DB_SERVER_NAME: '' # TODO: Fill in the name of the database server
+  DB_SERVER_PORT: '1433' # TODO: Verify that this is the correct port
+  DB_NAME: '' # TODO: Fill in the name of the database
+  MIGRATIONS_FOLDER: '' # TODO: Add the path to the migrations folder
+  MIGRATIONS_HISTORY_TABLE: 'MigrationHistory' # TODO: Verify this table name
+  TIMEZONE: 'america/denver' # TODO:  Verify timezone
+
+jobs:
+  set-vars:
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
+
+    outputs:
+      ENVIRONMENT: ${{ steps.clean-env.outputs.mapped_input }} # To use this value: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+
+    steps:
+      # To use this value: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+      - name: Set ENVIRONMENT
+        id: clean-env
+        uses: im-open/map-input-action@v1.0.1
+        with:
+          input: 'dev' #${{ github.event.inputs.environment }}
+          # TODO:  Update for the environments your project contains
+          # The value array contains the environments it will match and the corresponding key is
+          # the environment it will output if one of the values was found.  It is case insensitive.
+          input_map: |
+            { 
+              "dev": ["dev", "d", "development"], 
+              "qa": ["qa", "q"], 
+              "stage": ["stg", "s", "stage"], 
+              "demo": ["o", "demo"], 
+              "uat": ["u", "uat"], 
+              "prod": ["prod", "production", "p"] 
+            }
+          error_on_no_match: true
+          custom_error_message: 'The environment must be Dev, QA, Stage Demo, UAT or Prod'
+
+      - run: echo "The current environment is ${{ steps.clean-env.outputs.mapped_input }}"
+
+  # Each env has their own stakeholder approval environment.  If no required reviewers are set for
+  # that environment, the workflow will continue without requiring anyone to approve the deployment.
+  stakeholder-approval:
+    needs: [set-vars]
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
+    environment: '${{ needs.set-vars.outputs.ENVIRONMENT }} Stakeholder Approval'
+    steps:
+      - run: echo "The current environment is ${{ needs.set-vars.outputs.ENVIRONMENT }}"
+      - name: Approval Received
+        run: echo "Stakeholder approval was received"
+
+  # This job needs to run for all environments because deploy-az-db relies
+  # on it but the steps inside this job will only run for the Prod env.
+  validate-tag-for-prod-deploys:
+    needs: [set-vars, stakeholder-approval]
```

This potentially needs to be updated with attestor-approval too if you include it above
```suggestion
    needs: [set-vars, stakeholder-approval, attestor-approval]
```